python/pyspark/sql/avro/functions.py

```diff
@@ -25,7 +25,6 @@
 from pyspark.util import _print_missing_jar
 
 
-@since(3.0)
```

Comment:
`@since` just adds `.. versionadded::` at the end of the docstring, and this removal is necessary in some places, for example:

```
See Also
---------
blahblah
.. versionadded:: 3.0.0
```

causes NumPy docs style format error. Both `.. versionadded:: 3.0.0` and `@since` are used interchangeably in the codebase so I guess it's not a big deal.

Reply:
Does it mean we're going to effectively drop `@since` decorator as a side effect of SPARK-32085?

Reply:
When the docstring is compatible with both reST and NumPy doc style, we cna still use `@since` way .. but yes I am currently thinking we might just better remove it eventually.

Reply:
I've been looking at numpydoc documentation and it seems like the preferred order of sections is

> 1 Short summary
> 2 Deprecation warning
> 3 Extended Summary
> ...
> 13 Notes
> ...
> 15 Examples

with no dedicated section for `versionadded`.

If so, shall we standardize things a bit and try something around

1. Short summary
2. Extended Summary (optional)
3. Version added (optional)
4. Deprecation warning (optional)
...

This should match `@since`-annotated code and make things consistent across documentation (at the moment `New in version` is sometimes placed after examples and doesn't look good in the rendered docs).

Reply:
Sounds like a good lead to follow!

Reply:
I will add this link in SPARK-33256

Reply:
Also, it seems like `since` injected `versionadded` can render incorrectly, even if there is no format error. For example `Transformer.transform` renders like this, when decorator is used.

![Transformer.transform](https://user-images.githubusercontent.com/1554276/98161970-e8ab9a80-1ee0-11eb-92bf-4af57639de28.png)